design/private-repos.md

```diff
@@ -0,0 +1,47 @@
+# Screwdriver Access to Private Repositories
+
+## Context
+Provide a way for users to create and execute a pipeline for a private GitHub repository.
+
+This documentation covers the design proposal for that new feature.
+
+## Status
+
+- Oct 10, 2018: Design document
+
+## The Problem
+
+The current authentication methods do not provide access to private repositories.  
+Also, there are several stages of a pipeline in which different access keys are needed 
+when dealing with private repositories.
+This document will outline each stage and how it will be addressed.
+
+## Background
+
+Reference [Issue 1079][issue1079]
+
+## Proposal
+
+Pipeline creation:
+- User must provide a GitHub Deploy Key
+- User must provide a GitHub Personal Access Token
```

Comment:
Can this be the github Oauth token that we already store in the user table?

Reply:
This will need to be a token that has elevated privileges for access to private repos.  The current token only allows public repos.